### PR TITLE
Added tests for Get-Module over remoting

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Some additional tests that are remoting specific based on code coverage

Part of https://github.com/PowerShell/PowerShell/issues/4161

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        if ($parameter -eq "FullyQualifiedName") {
+            $getModuleCommand.FullyQualifiedName | Should BeOfType "Microsoft.PowerShell.Commands.ModuleSpecification"
+            $getModuleCommand.FullyQualifiedName.Name | Should Be "foo"
+            $getModuleCommand.FullyQualifiedName.Version | Should Be "1.2.3"
+        } else {
+            $getModuleCommand.$parameter | Should Be $value
+        }
+    }
+
+    It "ArugmentCompleter for PSEdition should work for '<cmd>'" -TestCases @(
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1]: I see currently we use `TabExpansion2 ` only in two files. Should we put completion tests in one place or put tab completion tests in cmdlet test files?
Make sense make one meta test to cover all (cmdlets) tab completions? It seems something like we do for aliases.

File: test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+
+    It "Get-Module fails if not using -ListAvailable with '<parameter>'" -TestCases @(
+        @{parameter="PSSession" ; value=$pssession},
+        @{parameter="CIMSession"; value=$cimsession}
+    ) {
+        param($parameter, $value)
+        $parameters = @{$parameter=$value}
+        { Get-Module @parameters } | ShouldBeErrorId "RemoteDiscoveryWorksOnlyInListAvailableMode,Microsoft.PowerShell.Commands.GetModuleCommand"
```
[the remark this answers is not in the document]
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1]: Will add

File: test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1]: I'll move this to TabCompletion.Tests.ps1

File: test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        $originalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if (!$IsWindows) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        } else {
+            $pssession = New-RemoteSession
+            $cimsession = New-RemoteSession -CimSession
+        }
+    }
+
+    AfterAll {
```
Comment (TravisEz13) [inline test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1]: Edge case (minor): Last time I checked, afterAll doesn't run if you hit control-c or some other error fatal to pester happens while running these tests.  The Try-Finally pattern might be better for setting up this parameter to ensure that it always gets removed.

**Review by TravisEz13 — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@SteveL-MSFT can you please add `[feature]` tag and trigger the feature test run? The full build triggered by the first commit failed.

File: test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.PowerShell.Core/RemoteGetModule.Tests.ps1]: We've been using this pattern everywhere.  If there is a case where AfterAll/AfterEach doesn't get run, we should open an issue for Pester

**Comment by SteveL-MSFT** [comment]
@daxian-dbw done

**Comment by daxian-dbw** [comment]
@SteveL-MSFT there is one failure in the new tests. Please fix the test.
```
Describing Remote module tests
VERBOSE: Using Global AppVeyor Credential
VERBOSE: Using Global AppVeyor Credential
 [-] Error occurred in Describe block 425ms
   Method not found: 'Void System.Security.SecureStringMarshal.ZeroFreeCoTaskMemUnicode(IntPtr)'.
   At C:\projects\powershell\test\tools\Modules\HelpersRemoting\HelpersRemoting.psm1:66 char:20
```
Please also include `[Feature]` tag when submitting new commits.

**Comment by SteveL-MSFT** [comment]
@daxian-dbw that api doesn't exist in our sources.  I suspect it's in MI.dll.

**Comment by daxian-dbw** [comment]
@SteveL-MSFT the error seems to be caused by `$cimsession = New-RemoteSession -CimSession`. It's thrown from `HelpersRemoting.psm1`. It possibly reveals a code bug.

**Comment by SteveL-MSFT** [comment]
I'm pretty sure it's a product issue.  However, it's likely an issue with mi.dll.

**Comment by daxian-dbw** [comment]
Can you please open an issue with a simple repro step? The tests that require `$CimSession` need to be marked pending.

**Comment by SteveL-MSFT** [comment]
Created https://github.com/PowerShell/PowerShell/issues/4819

**Comment by daxian-dbw** [comment]
There are 2 test failurs in AppVeryor test run. Can you please fix them?

**Pull request merged**